### PR TITLE
Remaining pages and auth

### PR DESCRIPTION
**Pull request by hiteshverma27**
- Added login, signup, watch later, playlist, profile pages
- Added auth. login signup and guest login

Deploy preview
https://62814986151db56e7c58e8df--subtle-syrniki-8d440e.netlify.app/

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,16 +1,38 @@
...
+  Home,
+  LikedVideos,
+  Login,
+  NotFound,
+  PlayList,
+  Signup,
+  VideoListing,
+  WatchLater,
+} from "./pages";
+import { Profile } from "./pages/Profile/Profile";
```
Comment (bhavyaj12) [inline src/App.js:16]: I guess Profile page can also be imported from the line above

File: src/components/VideoCard/VideoCard.css [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    min-height: 100%;
+    flex-shrink: 0;
+    border-radius: 1rem;
+}
+.img-container{
+    overflow: hidden;
+}
+.overflow-title{
+    white-space: nowrap;
+  overflow: hidden;
```
Comment (bhavyaj12) [inline src/components/VideoCard/VideoCard.css:19]: Be consistent with formatting and indentation

File: src/pages/Profile/Profile.js [hunk 3]
```diff
@@ -0,0 +1,63 @@
+import React from "react";
+import { NavBar, successToast } from "../../components";
+import { Link, useNavigate } from "react-router-dom";
+import { useAuth } from "../../contexts";
+
+function Profile() {
```
Comment (bhavyaj12) [inline src/pages/Profile/Profile.js:6]: Nice that you created a Profile page

File: src/pages/Signup/Signup.css [hunk 4]
```diff
@@ -0,0 +1,8 @@
+.color-green{
+    color: green;
```
Comment (bhavyaj12) [inline src/pages/Signup/Signup.css:2]: You should use color variables if possible and for font weight, use a numerical value

File: src/pages/Signup/Signup.js [hunk 5]
```diff
@@ -0,0 +1,205 @@
+import React, { useState } from "react";
+import { errorToast, NavBar, successToast } from "../../components";
```
Comment (bhavyaj12) [inline src/pages/Signup/Signup.js:2]: A cleaner coding practice would be to have all the global imports at the top, followed by local imports, so components one can come below with contexts

**Review by bhavyaj12 — APPROVED** [review]
Great work, code is nice and logic is well written. In terms of UI slight improvement can be for the toast to show at the top, not bottom. 

Have left some minor suggestions, other than that good to merge!

**Comment by hiteshverma27** [comment]
> Great work, code is nice and logic is well written. Have left some minor suggestions, other than that good to merge!

Hey Bhavya, thank you so much for taking the time to review this PR thoroughly. I will make the suggested changes in the next PR.

**Review by aniketsingh13 — APPROVED** [review]
code structure look good
good to merge

**Pull request merged**